activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -115,7 +115,16 @@ def load_records(&block)
             # Make several smaller queries if necessary or make one query if the adapter supports it
             slices = owner_keys.each_slice(klass.connection.in_clause_length || owner_keys.size)
             @preloaded_records = slices.flat_map do |slice|
-              records_for(slice).load(&block)
+              records = records_for(slice)
```

Comment:
> `records_for(slice, &block)`

:+1:

That allows the custom preloader to pass the block along to the load, so it doesn't break callbacks. It also eliminates a conditional -- one that would exist solely to handle this private-API-based use case.

This change, on its own, I'm okay with: it's arguably a better separation of responsibilities anyway.